### ISSUE TITLE
[4.0.7][Bug][CoreUI] Unable to trigger checkbox field by clicking on label

### ISSUE THREAD

**Issue by drionix** [Laravel-Backpack/CRUD#2140]
# Bug report

### What I did
Use the checkbox field
```
$this->crud->addField([
     'name' => 'active',
     'label' => 'Active',
     'type' => 'checkbox',
]);
```

### What I expected to happen

The checkbox is triggered by clicking on the label like on Backpack v3.6

### What happened

Clicking the label did not trigger the checkbox

### Possible Solution

Modify `checkbox.blade.php` field from
```
    <div class="checkbox">
        <input type="hidden" name="{{ $field['name'] }}" value="{{ old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? 0 }}">
    	  <input type="checkbox"
          data-init-function="bpFieldInitCheckbox"

          @if (old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? false)
                 checked="checked"
          @endif

          @if (isset($field['attributes']))
              @foreach ($field['attributes'] as $attribute => $value)
    			{{ $attribute }}="{{ $value }}"
        	  @endforeach
          @endif
          >
    	<label class="form-check-label font-weight-normal" for="{{ $field['name'] }}_checkbox">{!! $field['label'] !!}</label>

        {{-- HINT --}}
        @if (isset($field['hint']))
            <p class="help-block">{!! $field['hint'] !!}</p>
        @endif
    </div>
```
to
```
<div class="checkbox">
    <label class="form-check-label font-weight-normal">
	<input type="hidden" name="{{ $field['name'] }}" value="{{ old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? 0 }}">
	<input type="checkbox"
		data-init-function="bpFieldInitCheckbox"

		@if (old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? false)
			 checked="checked"
		@endif

		@if (isset($field['attributes']))
			 @foreach ($field['attributes'] as $attribute => $value)
				{{ $attribute }}="{{ $value }}"
			@endforeach
		@endif
	> {!! $field['label'] !!}
    </label>

    {{-- HINT --}}
    @if (isset($field['hint']))
         <p class="help-block">{!! $field['hint'] !!}</p>
    @endif
</div>
```

### Backpack, Laravel, PHP, DB version
Backpack 4.0.7
Laravel 6.2
PHP 7.2

When I run ```php artisan backpack:version``` the output is:

**Comment by johnpuddephatt** [comment]
Bootstrap suggests keeping the label and input as siblings. I'm not entirely sure I follow their reasoning but they say:

> Checkboxes and radios use are built to support HTML-based form validation and provide concise, accessible labels. As such, our `<input>`s and `<label>`s are sibling elements as opposed to an `<input>` within a `<label>`.

Regardless I'd suggest keeping to their suggested structure. That means the fix is making the `for` attribute on the label match the `id` attribute on the checkbox. This is fairly simple but I guess we need to handle the case where people manually provide an id through the attributes array?

I've created #2144 which I _think_ does this.

**Comment by tabacitu** [comment]
Thank you @drionix for the comprehensive bug report, and both you & @johnpuddephatt for the PRs. Indeed this was a bug - clicking the label should definitely trigger the checkbox. 

I've just merged @johnpuddephatt 's PR since seems like the least invasive and most "Bootstrap-recommended" solution. It'll be available later today as ```4.0.9```, and installable with a ```composer update```.

Thanks again.
Cheers!

**Comment by drionix** [comment]
Indeed I agree @johnpuddephatt solution is better.

@tabacitu And for the sake of consistency, I think `checklist.blade.php` and `checklist_dependency.blade.php` need to adjust with this change too.
